pkg/executables/cmk.go

```diff
@@ -265,22 +264,22 @@ func (c *Cmk) ValidateNetworkPresent(ctx context.Context, domainId string, zoneR
 	networks := response.CmkNetworks
 
 	// filter by network name -- cmk does not support name= filter
-	if zoneRef.Network.Type == v1alpha1.Name {
+	if len(zone.Network.Name) > 0 {
```

Comment:
Either Network name and ID must have at least one specified. The code is different from other cmk is because cloudstack cmk does not support list networks -name=, but support list networks -id.

So if we both name and ID are specified, or only name is specified, we have run  cmk list netwworks and filter the result to confirm we have one and only one network with that name.